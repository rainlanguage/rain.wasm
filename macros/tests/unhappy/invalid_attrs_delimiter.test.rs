#[macro_use]
extern crate wasm_bindgen_utils_macros;

struct TestStruct;

#[wasm_export]
impl TestStruct {
    #[wasm_export(skip; unchecked_return_type = "string")]
    pub async fn some_static_method1(arg: String) -> Result<String, Error> {
        Ok(String::new())
    }
}

#[wasm_export]
impl TestStruct {
    #[wasm_export(skip - unchecked_return_type = "string")]
    pub async fn some_static_method2(arg: String) -> Result<String, Error> {
        Ok(String::new())
    }
}

#[wasm_export]
impl TestStruct {
    #[wasm_export(skip. unchecked_return_type = "string")]
    pub async fn some_static_method3(arg: String) -> Result<String, Error> {
        Ok(String::new())
    }
}

#[wasm_export]
impl TestStruct {
    #[wasm_export(skip/ unchecked_return_type = "string")]
    pub async fn some_static_method4(arg: String) -> Result<String, Error> {
        Ok(String::new())
    }
}

#[wasm_export]
impl TestStruct {
    #[wasm_export(skip unchecked_return_type = "string")]
    pub async fn some_static_method5(arg: String) -> Result<String, Error> {
        Ok(String::new())
    }
}

fn main() {}