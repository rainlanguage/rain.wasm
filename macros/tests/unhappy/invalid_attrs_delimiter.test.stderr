error: expected `,` as wasm_export attributes must be delimited by comma
 --> tests/unhappy/invalid_attrs_delimiter.test.rs:8:23
  |
8 |     #[wasm_export(skip; unchecked_return_type = "string")]
  |                       ^

error: expected `,` as wasm_export attributes must be delimited by comma
  --> tests/unhappy/invalid_attrs_delimiter.test.rs:16:24
   |
16 |     #[wasm_export(skip - unchecked_return_type = "string")]
   |                        ^

error: expected `,` as wasm_export attributes must be delimited by comma
  --> tests/unhappy/invalid_attrs_delimiter.test.rs:24:23
   |
24 |     #[wasm_export(skip. unchecked_return_type = "string")]
   |                       ^

error: expected `,` as wasm_export attributes must be delimited by comma
  --> tests/unhappy/invalid_attrs_delimiter.test.rs:32:23
   |
32 |     #[wasm_export(skip/ unchecked_return_type = "string")]
   |                       ^

error: expected `,` as wasm_export attributes must be delimited by comma
  --> tests/unhappy/invalid_attrs_delimiter.test.rs:40:24
   |
40 |     #[wasm_export(skip unchecked_return_type = "string")]
   |                        ^^^^^^^^^^^^^^^^^^^^^
